mindmap/tests/test_mindmap.py

```diff
@@ -0,0 +1,241 @@
+        self.xblock.resource_string = Mock()
+        self.xblock.display_name = "Test MindMap"
+
+    @patch("mindmap.mindmap.Fragment.initialize_js")
+    def test_student_view_with_mind_map(self, initialize_js_mock):
+        """
+        Check student view is rendered correctly.
+
+        Expected result:
+            - The student view is set up for the render with the student.
+        """
+        mind_map = {"data": "content"}
+        self.xblock.is_student.return_value = True
+        self.xblock.anonymous_user_id.return_value = self.anonymous_user_id
+        self.xblock.get_current_user.return_value = self.student
+        self.xblock.get_current_mind_map.return_value = mind_map
+        context = {"show_mindmap": True, "error_message": None}
+        expected_js_context = {"mind_map": mind_map, "author": self.student.full_name}
+
+        self.xblock.student_view()
+
+        initialize_js_mock.assert_called_once_with(
+            'MindMapXBlock', json_args=expected_js_context
+        )
+        self.xblock.render_template.assert_called_once_with(
+            "static/html/mindmap.html", context
+        )
+
+    @patch("mindmap.mindmap.Fragment.initialize_js")
+    def test_student_view_empty_mind_map(self, initialize_js_mock):
+        """
+        Check student view is rendered correctly.
+
+        Expected result:
+            - The student view is set up for the render with the student.
+        """
+        self.xblock.is_student.return_value = True
+        self.xblock.get_current_mind_map.return_value = None
+        self.xblock.get_current_user.return_value = self.student
+        self.xblock.anonymous_user_id.return_value = self.anonymous_user_id
+        context = {"show_mindmap": True, "error_message": None}
+        expected_js_context = {"mind_map": None, "author": self.student.full_name}
+
+        self.xblock.student_view()
+
+        initialize_js_mock.assert_called_once_with(
+            'MindMapXBlock', json_args=expected_js_context
+        )
+        self.xblock.render_template.assert_called_once_with(
+            "static/html/mindmap.html", context
+        )
+
+    @patch("mindmap.mindmap.Fragment.initialize_js")
+    def test_student_view_not_show_mind_map(self, initialize_js_mock):
+        """
+        Check student view is rendered correctly when called from the Studio.
+
+        Expected result:
+            - The mind map is not set by the handler.
+            - A message is rendered instead.
+        """
+        self.xblock.is_student.return_value = False
+        self.xblock.user_is_staff.return_value = False
+        self.xblock.get_current_mind_map.return_value = None
+        self.xblock.get_current_user.return_value = self.student
+        self.xblock.anonymous_user_id.return_value = self.anonymous_user_id
+        context = {"show_mindmap": False, "error_message": None}
+        expected_js_context = {"author": self.student.full_name}
```

Comment:
What if the student doesn't have a full name? should we use the username instead -- students always have a username? This comment was something I missed from the implementation. We can work on it in another PR. But please, don't forget :)